cores/esp32/esp32-hal-gpio.c

```diff
@@ -182,14 +182,12 @@ extern int ARDUINO_ISR_ATTR __digitalRead(uint8_t pin) {
   if (pin == RGB_BUILTIN) {
     return RGB_BUILTIN_storage;
   }
-#endif
-
+#endif  // RGB_BUILTIN
+  // This work when the pin is set as GPIO and in INPUT mode. For all other pin functions, it may return inconsistent response
   if (perimanGetPinBus(pin, ESP32_BUS_TYPE_GPIO) != NULL) {
```

Comment:
This should check if getPinBus is equal to NULL, else you are sending warning when the pin is actually set as GPIO

Reply:
Correct.